build.ps1

```diff
@@ -53,6 +53,125 @@ if ($Clean -and (Test-Path "$PSScriptRoot/bin")) {
     Remove-Item "$PSScriptRoot/bin" -Recurse -Force
 }
 
+if ($Clean) {
+    # Import-LocalizedData (and ModuleVersion-property) used as workaround due to unknown error on PS3 build with Test-ModuleManifest
+    # and because Test-ModuleManifest needs the psd1 and psm1 to be complete, but we want to generate help for config from the type
+    # so we need to build up here, and not after the module build, so xml based solution is better than one that validates the manifest
+    $manifest = Import-LocalizedData -FileName "Pester.psd1" -BaseDirectory "$PSScriptRoot/src"
+    dotnet build "$PSScriptRoot/src/csharp/Pester.sln" --configuration Release -p:VersionPrefix="$($manifest.ModuleVersion)" -p:VersionSuffix="$($manifest.PrivateData.PSData.Prerelease)"
+    if (0 -ne $LASTEXITCODE) {
+        throw "build failed!"
+    }
+}
+
+function Copy-Content ($Content) {
+    foreach ($c in $content) {
+        $source, $destination = $c
+
+        $null = New-Item -Force $destination -ItemType Directory
+
+        Get-ChildItem $source -File | Copy-Item -Destination $destination
+    }
+}
+
+$content = @(
+    , ("$PSScriptRoot/src/en-US/*.txt", "$PSScriptRoot/bin/en-US/")
+    , ("$PSScriptRoot/src/nunit_schema_2.5.xsd", "$PSScriptRoot/bin/")
+    , ("$PSScriptRoot/src/junit_schema_4.xsd", "$PSScriptRoot/bin/")
+    , ("$PSScriptRoot/src/report.dtd", "$PSScriptRoot/bin/")
+    , ("$PSScriptRoot/src/Pester.ps1", "$PSScriptRoot/bin/")
+    , ("$PSScriptRoot/src/Pester.psd1", "$PSScriptRoot/bin/")
+)
+
+if ($Clean) {
+    $content += @(
+        , ("$PSScriptRoot/src/csharp/Pester/bin/Release/net452/Pester.dll", "$PSScriptRoot/bin/bin/net452/")
+        , ("$PSScriptRoot/src/csharp/Pester/bin/Release/net452/Pester.pdb", "$PSScriptRoot/bin/bin/net452/")
+        , ("$PSScriptRoot/src/csharp/Pester/bin/Release/netstandard2.0/Pester.dll", "$PSScriptRoot/bin/bin/netstandard2.0/")
+        , ("$PSScriptRoot/src/csharp/Pester/bin/Release/netstandard2.0/Pester.pdb", "$PSScriptRoot/bin/bin/netstandard2.0/")
+    )
+}
+
+Copy-Content -Content $content
+
+
+# update help for New-PesterConfiguration
+if ($PSVersionTable.PSVersion.Major -gt 5) {
+    Import-Module "$PSScriptRoot/bin/bin/netstandard2.0/Pester.dll"
+}
+else {
+    Import-Module "$PSScriptRoot/bin/bin/net452/Pester.dll"
+}
```

Comment:
@nohwnd This causes the failures. Needs cleanup after as it causes the dll to be shown as a binary module in all p-tests since they invoke `& "$PSScriptRoot\..\build.ps1"` at the beginning. So later when you build the path for the schema, you get paths for both "modules":

```
ModuleType Version PreRelease Name   ExportedCommands
---------- ------- ---------- ----   ----------------
Script     5.2.0   rc1        Pester {Add-ShouldOperator, AfterAll, AfterEach, Assert-MockCalled…}
Binary     5.2.0.0            Pester 
```

Reply:
Ah! Need to run the doc generation in Job in that case (probably better but might be slow). Or load it via assembly.loadfrom. Thanks!

Reply:
Do jobs have their own appdomain? If not, then the assembly will be loaded in the powershell session either way, so could just do `Remove-Module`  after doc generation to avoid the module-metadata conflict later in the tests. Or would e.g. a job give other benefits?